services/web/server/src/simcore_service_webserver/publication_handlers.py

```diff
@@ -0,0 +1,60 @@
+import json
+import os
+import logging
+
+from aiohttp import MultipartReader, hdrs, web
+from json2html import json2html
+
+from .login.decorators import RQT_USEREMAIL_KEY, login_required
+
+log = logging.getLogger(__name__)
+
+support_email_address = 'support@osparc.io'
+email_template_name = 'service_submission.html'
+
+@login_required
+async def service_submission(request: web.Request):
+    reader = MultipartReader.from_response(request)
+    data = None
+    filedata = None
+    # Read multipart email
+    while True:
+        part = await reader.next()
+        if part is None:
+            break
+        if part.headers[hdrs.CONTENT_TYPE] == 'application/json':
+            data = await part.json()
+            continue
+        if part.headers[hdrs.CONTENT_TYPE] == 'application/zip':
+            filedata = await part.read(decode=True)
+            # Validate max file size
+            maxsize = 10 * 1024 * 1024 # 10MB
+            actualsize = len(filedata)
+            if actualsize > maxsize:
+                raise web.HTTPRequestEntityTooLarge(maxsize, actualsize)
+            filename = part.filename
+            continue
+        raise web.HTTPUnsupportedMediaType(reason=f'One part had an unexpected type: {part.headers[hdrs.CONTENT_TYPE]}')
+    # data (dict) and file (bytearray) have the necessary information to compose the email
+    user_email = request.get(RQT_USEREMAIL_KEY)
+    try:
+        from .login.utils import common_themed, render_and_send_mail
+        # send email
+        subject = 'New service submission'
+        is_real_usage = any([env in os.environ.get("SWARM_STACK_NAME") for env in ('production', 'staging')])
```

Comment:
MINOT: why not to add this check as a function in  ``servicelib`` so we centralize this logic? This way any service can determine which deploy is it in. Notice that SWARM_STACK_NAME is a standard variable osparc-wide

Reply:
sounds very good but I might need some introduction about ``servicelib``

Reply:
i mean just to add that call in some utils in ``serviceslib``